tests/src/main/java/com/hazelcast/stabilizer/tests/helpers/HazelcastTestUtils.java

```diff
@@ -167,16 +167,32 @@ public static HazelcastInstanceImpl getHazelcastInstanceImpl(HazelcastInstance h
         return impl;
     }
 
-    public static long nextKeyOwnedBy(long key, HazelcastInstance instance) {
-        final Member localMember = instance.getCluster().getLocalMember();
-        final PartitionService partitionService = instance.getPartitionService();
-        for (; ; ) {
+    /**
+     * Returns the next {@code long} key owned by the given Hazelcast instance.
+     * <p/>
+     * If no key is found after 100 retries a {@link RuntimeException} is thrown.
+     *
+     * @param instance  Hazelcast instance to search next key for
+     * @param lastKey   last key to start search from
+     * @param increment increment to add if next key did not belong to given instance
+     * @return next key owned by given Hazelcast instance
+     * @throws RuntimeException if no key could be found after 100 retries
+     */
+    public static long nextKeyOwnedBy(HazelcastInstance instance, long lastKey, int increment) {
+        if (increment < 1) {
+            throw new IllegalArgumentException("increment must be at least 1");
+        }
+        Member localMember = instance.getCluster().getLocalMember();
+        PartitionService partitionService = instance.getPartitionService();
+        long key = lastKey;
+        for (int retries = 0; retries < 100; retries++) {
```

Comment:
I personally like the loop for ever trick for this, yes it loops for ever but,  in reality it should not.
also 100 is hard coded, and a very small number of retries.  if we have a 256 node cluster,  I might not own any keys in the range 0 to 100